app/models/concerns/user_session_attributes.rb

```diff
@@ -0,0 +1,49 @@
+module UserSessionAttributes
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def attribute(attribute, answer: true)
+      define_method(attribute) do
+        return session['answers'][attribute.to_s] if answer
+
+        session[attribute.to_s]
+      end
+
+      define_method("#{attribute}=") do |value|
+        return session['answers'][attribute.to_s] = value if answer
+
+        session[attribute.to_s] = value
+      end
+    end
+
+    def dual_route_attribute(attribute)
```

Comment:
Nice idea here!
the name was a bit strange at first, it's like route reminded me of the routing,
maybe we could be more explicit, maybe something like  `attribute_for_uk_and_xi`